pages/glossary/content-that-becomes-visible.md

```diff
@@ -0,0 +1,25 @@
+---
+title: Content that becomes visible
+key: content-that-becomes-visible
+unambiguous: true
+objective: true
+input_aspects:
+  - CSS styling
+  - DOM tree
+---
+
+The _content that becomes visible_ is the [root][] element of the [tree][], if a [tree][] exists, that contains all elements that, after half a second of an [event][] [firing][], meet any of the following:
+
+- the element did not exist on the [web page][] prior to the [event][] [firing][] and has [visible text content][]; or
```

Comment:
I think there is a fundamental problem in this. If I have an event rerenders the page, all elements are new, even if nothing changed. `body.innerHTML = body.innerHTML` will do that. Modern frameworks try to avoid unnecessary rerenders, but there is a lot of code on the web that rerenders components, irrespective of if something actually changed.

I think we need to define this in a way that works with rerenders. I took a stab at doing that just for text nodes, here's what I've come up with. That isn't enough, but I hope it gets helps move this forward.


```md
A text node is considered to be _visible after an event_ if both of the following of the following are true:

- A tick before the event, there was no [visible][] text node with the same node value; and
- 500 milliseconds after the [event][] [firing][], there the text node is [visible][].
```

Reply:
I don't think that works because it requires that there are no text nodes with the same node value. If the new content includes a text node with the same value of an already present text node, then it will miss the new content.
I'll try to think of a solution for this...

Reply:
Following your suggestion, check if you think that what I came up with works